CMakeLists.txt

```diff
@@ -8,14 +8,27 @@ set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_CXX_STANDARD_REQUIRED True)
 set(CMAKE_RUNTIME_OUTPUT_DIRECTORY bin)
 
-add_definitions(-Wall -Wextra --pedantic)
-add_definitions(-DNVRTC_GET_TYPE_NAME=1)
+if(CMAKE_BUILD_TYPE STREQUAL "Debug")
+    add_definitions(-Dcurrent_log_level=loglevel::DEBUG)
+endif()
+
+if(CMAKE_COMPILER_IS_GNUCXX)
+    add_definitions(--pedantic -Wextra)
+elseif(CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
+    set(WINDOWS_EXPORT_ALL_SYMBOLS TRUE)
+endif()
+add_definitions(-DNVRTC_GET_TYPE_NAME=1 -Wall)
 
 include_directories(include)
 include_directories(include/yacx)
 include_directories(${CMAKE_CUDA_TOOLKIT_INCLUDE_DIRECTORIES})
 
-link_directories("${CMAKE_CUDA_TOOLKIT_INCLUDE_DIRECTORIES}/../lib")
+if (CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
+    link_directories("${CMAKE_CUDA_TOOLKIT_INCLUDE_DIRECTORIES}/../lib/x64")
```

Comment:
looks weird, but works? not sure why this is necessary since `link_libraries `should be enough and is enough for `nvrtc`